extensions/warp-mp-ipfs/src/store/friends.rs

```diff
@@ -385,21 +386,28 @@ impl<T: IpfsTypes> FriendsStore<T> {
     async fn check_queue(&self) -> anyhow::Result<()> {
         let list = self.queue.read().clone();
         for item in list.iter() {
-            let Queue(peer, data) = item;
+            let Queue(peer, data, recipient) = item;
             if let Ok(crate::store::PeerConnectionType::Connected) =
                 connected_to_peer(self.ipfs.clone(), *peer).await
             {
                 let bytes = serde_json::to_vec(&data)?;
 
-                self.ipfs
-                    .pubsub_publish(FRIENDS_BROADCAST.into(), bytes)
-                    .await?;
+                let sender = data.clone().sender().unwrap();
+
+                let recipients = match data.recipients() {
+                    Some(rec_vec) => {rec_vec.into_iter().filter(move |rec| *rec.fingerprint() != sender.fingerprint() ).collect::<Vec<KeyPair>>()},
```

Comment:
we could simply this by doing something like

```rust
data.recipients().map(|list| list.iter().filter(move |rec| *rec.fingerprint() != sender.fingerprint() ).collect::<Vec<_>>()).unwrap_or_default()
```

though i dont see this being used anyplace and probably wouldnt be needed